debian/kolibri-server.init

```diff
@@ -94,20 +104,18 @@ do_check_nginx()
   service nginx status 1>/dev/null 2>&1
   retval=$?
   if [ $retval -ne 0  ]; then
-    service kolibri restart
+    $SU_COMMAND $KOLIBRI_USER -c "$KOLIBRI_COMMAND restart"
     service nginx start
   else
     service nginx reload
   fi
 }
 
 do_stop() {
-  service kolibri stop
+  $SU_COMMAND $KOLIBRI_USER -c "$KOLIBRI_COMMAND stop"
```

Comment:
Why did you remove the `&` at the end of the command? locking causes important delays when restarting the associate processes. Same thing with the restart command.
It causes that, for example, restarting or halting the server takes an unneeded time while waiting for kolibri to shutdown the zeroconf daemons, what is unneeded both restarting or stopping.
Did you see any issue by adding it?

Reply:
Commented on that further up: https://github.com/learningequality/kolibri-server/pull/54#discussion_r393977573

Forking `kolibri stop` to the background and then restarting it simultaneously seems to be asking for unexpected behavior? I don't think it's worth it if -- theoretically -- it means more malformed databases or broken upgrades (restart is a part of upgrading, I think).

It could also basically just make `restart` broken, in cases where kolibri refuses to start because another instance still hasn't stopped?

Reply:
I understand it can make stopping the service faster.. but https://github.com/learningequality/kolibri-installer-debian/issues/90 has a promising fix for the perhaps most annoying part :)

Reply:
If https://github.com/learningequality/kolibri-installer-debian/issues/90 works we can skip this change. If not, I'd keep it at least for the `stop` part. That's what affects halting a machine and zeroconf does not matter at that moment. In the `restart` part we would have to live with the delay if we see it produces issues.

Reply:
I would feel a lot better about the `&` addition if it went through a stress test.. like a very fast system running `systemctl restart kolibri` 100 times. We should do that some day in any case, so once we get around to that, we can see if stopping can be made more concurrent?

Adding a follow-up issue: https://github.com/learningequality/kolibri-server/issues/62

Reply:
I understand your concern when using `restart` (and I agree to remove the `&` until we test it more) but I can't find a reason for problems when using `stop`, why do you think the service should be waiting for kolibri to stop instead of launching it in the background?

Reply:
Sorry I have been grasping this sporadically -- forget everything about `restart`, what worried me originally was breaking stuff during upgrades.

This holds true for the `kolibri` package, but I can't find out what the defined behavior is for `kolibri-server` package?

Here's what I feared could happen:

```
sudo apt upgrade

kolibri.preinst (debian script)
=> Invokes systemctl stop kolibri / service kolibri stop

   Time 0: "kolibri stop &"  <= forks into background
   Time 0: "start-stop-daemon uwsgi" <= blocks
   Time 1: UWSGI is now stopped

kolibri.postinst
=> Involes systemctl start kolibri / service kolibri start

   Time 2: "kolibri start"
   Time 3: "kolibri stop" finishes # called at Time 0
```

If apt/dpkg issues a service `restart` after the upgrade, then it's all safe. If it stops the service before the upgrade and starts it again after, I would consider it unsafe.